lib/TokenList.php

```diff
@@ -149,6 +149,115 @@ public function addTokensFromDB(&$oDB, &$aTokens, &$aCountryCodes, $sNormQuery,
         }
     }
 
+    /**
+     * Inspecting unmatched tokens check if another variation of postcod exist. A
+     * variation might be another normalization (e.g. needs space) or looking for
+     * the broader/shortened postcode.
+     *
+     * @param string[] $aTokens       List of tokens to look up in the database.
+     * @param string[] $aCountryCodes List of country restrictions.
+     *
+     * @return void
+     */
+    public function addAdditionalPostcodeTokens(&$aTokens, &$aCountryCodes)
+    {
+        foreach ($aTokens as $sToken) {
+            if ($sToken[0] == ' ') {
```

Comment:
You have left out the check for the token being unmatched: && !$oValidTokens->contains($sToken)

Reply:
added and added a test